catmap/api/cathub.py

```diff
@@ -0,0 +1,267 @@
+                                    '.us-west-2.rds.' +
+                                    'amazonaws.com:5432/catalysishub')
+        images = []
+        for uid in tqdm(unique_ids):
+            atoms = self.c.get_atoms(['unique_id=' + uid],
+                                     add_additional_information=True)
+            images.append(atoms)
+        return images
+
+    def get_publication_atoms(self, publication, limit=1):
+        """Return a list of ASE-db unique_id's from a catalysishub query.
+
+        Parameters
+        ----------
+        reaction : str
+            GraphQL search string. This search string for the publications
+            table. The string must be enclosed in three double hyphens and
+            additional double hyphens must be preceeded by a backslash.
+        """
+        if 'first' not in publication:
+            publication += """, first: """ + str(limit)
+        query = \
+            """{
+              publications(""" + publication + """) {
+                edges {
+                  node {
+                    systems {
+                      uniqueId
+                    }
+                  }
+                }
+              }
+            }
+            """
+
+        data = requests.post(self.root, {'query': query}).json()
+        if 'errors' in data:
+            raise KeyError(data['errors'])
+
+        unique_ids = []
+        for dat in data['data']['publications']['edges'][0]['node']['systems']:
+            unique_ids.append(str(dat['uniqueId']))
+
+        return self.get_atoms(unique_ids)
+
+    def import_energies(self, unique_ids, previous=None, site_specific=False):
+        """Return CatMAP energy_landscape object with potential energies.
+
+        Parameters
+        ----------
+        fname : str
+            Path and filename of candidate ase database file.
+        database_ids : list
+            Database ids.
+        prediction : list
+            Predicted means in the same order as database_ids.
+        uncertainty : list
+            Predicted uncertainties in the same order as database_ids.
+        catmap : object
+            CatMAP energy_landscape object.
+        site_specific : bool
+            If True: Dinstinguish sites using the site key value pair, and
+            stores a the potential energy of adsorbates on each site.
+            Else: Use the minimum ab initio energy, disregarding the site.
+        """
+        if previous is None:
+            energy_landscape = EnergyLandscape()
+        else:
+            energy_landscape = previous
+
+        if self.c is None:
+            self.c = ase.db.connect('postgresql://' +
+                                    self.username + ':' +
+                                    self.password +
+                                    '@catalysishub.c8gwuc8jwb7l' +
+                                    '.us-west-2.rds.' +
+                                    'amazonaws.com:5432/catalysishub')
```

Comment:
I'll attach them to the class in `__init__`. The user will then have the power to retrieve and update them in case the urls change for some reason, and not need to update the code itself.

Reply:
Sounds good.